CMakeLists.txt

```diff
@@ -758,8 +758,14 @@ if(CONFIG_BUILD_PROTECTED)
     OUTPUT_VARIABLE nuttx_user_libgcc)
 
   # reset link options for userspace to prevent sections from being accidentally
-  # deleted
-  set_target_properties(nuttx_user PROPERTIES LINK_OPTIONS "")
+  # deleted Toolchain link options
+  get_target_property(nuttx_user_LINK_OPTIONS nuttx_user LINK_OPTIONS)
+  list(REMOVE_ITEM nuttx_user_LINK_OPTIONS "-Wl,--gc-sections")
```

Comment:
why not skip adding these option instead

Reply:
> why not skip adding these option instead

These link options is set in the Toolchain file and has global scope.
they are also required by the nuttx target.but the timing of setting the Toolchain file is before all targets.
so it cannot be distinguished by USER_LINK_OPTION and NUTTX_LINK_OPTION like Makefile.

Reply:
I don't know how @xuxin930 identified them.  My test shows that the `gc-sections` will cause user space ELF lacking things required bykernel side, thus can't boot the target. The `-Map=nuttx.map` name conflicts with that of kernel. As these global settings are for kernel, adjusting them for userspace looks reasonable to me.

Reply:
@xuxin930 let's split the hack to a new pr